### PR TITLE
fix: Remove Node 10 support

### PR DESCRIPTION
**Pull request by ronag**
Fixes: https://github.com/nodejs/undici/issues/622

### CODE REVIEW

**Review by dnlup — APPROVED** [review]

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.16.0, 10.x, 12.3.0, 12.18.0, 12.x, 14.0.0, 14.x, 15.x]
+        node-version: [12.3.0, 12.18.0, 12.x, 14.0.0, 14.x, 15.x]
```
Comment (mcollina) [inline .github/workflows/nodejs.yml:16]:
```suggestion
        node-version: [12.x, 14.x, 15.x]
```

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you please update the minimum runtime in package.json to v12.18 (or 10.20)?

**Comment by ronag** [comment]
> Can you please update the minimum runtime in package.json to v12.18 (or 10.20)?

Where do I do this?

**Comment by dnlup** [comment]
> > Can you please update the minimum runtime in package.json to v12.18 (or 10.20)?
> 
> Where do I do this?

We should add the `engines` field: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.16.0, 10.x, 12.3.0, 12.18.0, 12.x, 14.0.0, 14.x, 15.x]
+        node-version: [12.x, 14.x, 15.x]
```
Comment (ronag) [inline .github/workflows/nodejs.yml:16]:
```suggestion
        node-version: [12.18, 14.x, 15.x]
```

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**